src/main/java/org/opensrp/web/rest/TaskResource.java

```diff
@@ -201,18 +206,25 @@ public void setTaskService(TaskService taskService) {
 
 	/**
 	 * This methods provides an API endpoint that searches for all task Ids
+	 * ordered by server version ascending
 	 *
+	 * @param serverVersion serverVersion using to filter by
 	 * @return A list of task Ids
 	 */
 	@RequestMapping(value = "/findIds", method = RequestMethod.GET, produces = {
 			MediaType.APPLICATION_JSON_VALUE })
-	public ResponseEntity<String> findIds() {
+	public ResponseEntity<Identifier> findIds(
+			@RequestParam(value = SERVER_VERSION)  long serverVersion) {
 
 		try {
-			return new ResponseEntity<>(
-					gson.toJson(taskService.findAllTaskIds()), HttpStatus.OK);
+			Pair<List<String>, Long> taskIdsPair = taskService.findAllTaskIds(serverVersion, DEFAULT_GET_ALL_IDS_LIMIT);
+			Identifier identifiers = new Identifier();
+			identifiers.setIdentifiers(taskIdsPair.getLeft());
+			identifiers.setLastServerVersion(taskIdsPair.getRight());
+
+			return new ResponseEntity<>(identifiers, HttpStatus.OK);
 		} catch (Exception e) {
```

Comment:
can we can catch something more specific?

Reply:
This catches all possible errors preventing leaking of stack-trace. This can be removed the global error handler is configured https://github.com/OpenSRP/opensrp-server-web/issues/245